### ISSUE TITLE
TestDataLoader.test_proper_exit takes 2.5min to finish

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#46065]
According https://app.circleci.com/pipelines/github/pytorch/pytorch/224304/workflows/2571002d-2075-4e53-877b-4bac675037de/jobs/8062979
```
TestDataLoader.test_proper_exit  time: 158.54 seconds
TestDataLoaderPersistentWorkers.test_proper_exit  time: 158.44 seconds
```

**Comment by heitorschueroff** [comment]
@SsnL is this a known issue? Can you advise on what may be causing this?

**Comment by SsnL** [comment]
Yeah the test is pretty slow, but it has many nested loops running
different configurations of the same setting so if you break it up it is
not too bad.
